File: historyscan-back/src/main/java/io/sch/historyscan/infrastructure/features/analysis/dto/network/NetworkNodeDTO.java
package io.sch.historyscan.infrastructure.features.analysis.dto.network;

import com.fasterxml.jackson.annotation.JsonCreator;

import java.util.List;
import java.util.Objects;

public class NetworkNodeDTO {
    private final String name;
    private final String path;
    private final String parentPath;
    private final long currentNbLines;
    private final long score;
    private final List<NetworkLinkDTO> links;

    @JsonCreator
    public NetworkNodeDTO(String name, String path, String parentPath, long currentNbLines, long score, List<NetworkLinkDTO> links) {
        this.name = name;
        this.path = path;
        this.parentPath = parentPath;
        this.currentNbLines = currentNbLines;
        this.score = score;
        this.links = links;
    }

    public String getName() {
        return name;
    }

    public String getPath() {
        return path;
    }

    public String getParentPath() {
        return parentPath;
    }

    public long getCurrentNbLines() {
        return currentNbLines;
    }

    public long getScore() {
        return score;
    }

    public List<NetworkLinkDTO> getLinks() {
        return links;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        NetworkNodeDTO that = (NetworkNodeDTO) o;
        return currentNbLines == that.currentNbLines
               && score == that.score
               && Objects.equals(name, that.name)
               && Objects.equals(path, that.path)
               && Objects.equals(parentPath, that.parentPath)
               && Objects.equals(links, that.links);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, path, parentPath, currentNbLines, score, links);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/infrastructure/features/analysis/dto/network/NetworkNodeDTO.kt
package io.sch.historyscan.infrastructure.features.analysis.dto.network

import com.fasterxml.jackson.annotation.JsonCreator
import java.util.*

class NetworkNodeDTO @JsonCreator constructor(val name: String, val path: String, val parentPath: String, val currentNbLines: Long, val score: Long, val links: List<NetworkLinkDTO>) {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as NetworkNodeDTO
        return currentNbLines == that.currentNbLines && score == that.score && name == that.name && path == that.path && parentPath == that.parentPath && links == that.links
    }

    override fun hashCode(): Int {
        return Objects.hash(name, path, parentPath, currentNbLines, score, links)
    }
}
